tools/topology/sof-tgl-rt711-rt1308.m4

```diff
@@ -58,6 +58,16 @@ ifdef(`EXT_AMP',
 '
 )
 
+# Add Bluetooth Audio Offload pass-through for ADL
+ifelse(PLATFORM, `adl', `
+	define(`BT_PIPELINE_PB_ID', `12')
+	define(`BT_PIPELINE_CP_ID', `13')
+	define(`BT_DAI_LINK_ID', eval(HDMI_BE_ID_BASE + 4))
+	define(`BT_PCM_ID', `10')
+	ifelse(CHANNELS,`0', `define(HW_CONFIG_ID, `6')', `define(HW_CONFIG_ID, `8')')
+	include(`platform/intel/intel-generic-bt.m4')
+', `')
+
```

Comment:
do you need to test for ADL or could this be added for TGL as well? Would this break anything?

Reply:
Yes, need to test for ADL. For TGL, we already had a ref implementation (max98373 + rt5682) for joint development purpose. I would prefer to prioritize ADL this time if possible, thank!!

Reply:
@yongzhi1 I have a ADL-P (rt711 or rt711+rt1316+rt715). If you need help to test on it, please let me know how to test.

Reply:
> Yes, need to test for ADL. For TGL, we already had a ref implementation (max98373 + rt5682) for joint development purpose. I would prefer to prioritize ADL this time if possible, thank!!
> 
> (Forget to mention, I made local changes [yongzhi1/linux@a58ad70](https://github.com/yongzhi1/linux/commit/a58ad705404712a3a26affd81a907131b1436455) to enable BT offload, will sync with @Vamshigopal to see any more entry in the m/c for ADL)

My question was: can we add this unconditionally for both ADL and TGL, or is there any machine driver dependency that would break on TGL?

Reply:
I'm doing tests today @yongzhi1 and @libinyang . @plbossart There is a link to the machine driver and we'd need to enable the BT node for TGL as well. Let us investigate this a bit more -- not sure yet what (if anything) this would break. At first, some platforms where Bluetooth audio offload is not available, would have a ALSA PCM still visible. OTOH, we have a PCM for SPDIF even if no connectors are available, so this is not unheard of.

Reply:
> My question was: can we add this unconditionally for both ADL and TGL, or is there any machine driver dependency that would break on TGL?

There is machine driver dependency:
- DAI link must be added in the m/c to match the BT offload BE id defined in topology.
- Another hard limitation is that this feature is only available on SSP2 only for TGL/ADL.

Reply:
Basic test successful and Bluetooth part works ok. There's some issue when SOF_SDW_PCH_DMIC is defined. I'm getting a failure in kernel spcm_bind() when loading the DMIC DAI. I'll debug more on Monday, but I think this looks ok to remove the draft status and proceed with the review.

@plbossart I think we are better off to leave TGL out for now. It will need a quirk update to sof_sdw.c and currently we have no immediate need to do this.

Reply:
Yes, see below, we use this topology both for an RVP and an HP platform:

````c
	{
		.link_mask = 0x3, /* rt711 on link 0 and 1 rt1308 on link 1 */
		.links = tgl_hp,
		.drv_name = "sof_sdw",
		.sof_tplg_filename = "sof-tgl-rt711-rt1308.tplg",
	},
	{
		.link_mask = 0x3, /* rt711 on link 0 and 2 rt1308s on link 1 */
		.links = tgl_rvp,
		.drv_name = "sof_sdw",
		.sof_tplg_filename = "sof-tgl-rt711-rt1308.tplg",
	},
````

I don't want to take the risk of changing this topology for TGL. It was painful enough to support this HP platform, I don't want to break it now. We could add the code in the machine driver with a quirk for the RVP, and have this topology changed to add BT offload in a development directory, but not for external sof-bin releases.